src/openrct2-ui/windows/TopToolbar.cpp

```diff
@@ -278,3593 +278,3549 @@ static Widget window_top_toolbar_widgets[] = {
+                screenPos.y++;
+            gfx_draw_sprite(&dpi, ImageId(SPR_FINANCE), screenPos);
+        }
 
-/**
- *
- *  rct2: 0x0066CD0C
- */
-static void ToggleClearSceneryWindow(rct_window* topToolbar, WidgetIndex widgetIndex)
-{
-    if ((input_test_flag(INPUT_FLAG_TOOL_ACTIVE) && gCurrentToolWidget.window_classification == WindowClass::TopToolbar
-         && gCurrentToolWidget.widget_index == WIDX_CLEAR_SCENERY))
-    {
-        tool_cancel();
-    }
-    else
-    {
-        show_gridlines();
-        tool_set(*topToolbar, widgetIndex, Tool::Crosshair);
-        input_set_flag(INPUT_FLAG_6, true);
-        ContextOpenWindow(WindowClass::ClearScenery);
+        // Draw news button
+        if (widgets[WIDX_NEWS].type != WindowWidgetType::Empty)
+        {
+            screenPos = windowPos + ScreenCoordsXY{ widgets[WIDX_NEWS].left + 3, widgets[WIDX_NEWS].top + 0 };
+            if (WidgetIsPressed(*this, WIDX_NEWS))
+                screenPos.y++;
+            gfx_draw_sprite(&dpi, ImageId(SPR_G2_TAB_NEWS), screenPos);
+        }
+
+        // Draw network button
+        if (widgets[WIDX_NETWORK].type != WindowWidgetType::Empty)
+        {
+            screenPos = windowPos + ScreenCoordsXY{ widgets[WIDX_NETWORK].left + 3, widgets[WIDX_NETWORK].top + 0 };
+            if (WidgetIsPressed(*this, WIDX_NETWORK))
+                screenPos.y++;
+
+            // Draw (de)sync icon.
+            imgId = (network_is_desynchronised() ? SPR_G2_MULTIPLAYER_DESYNC : SPR_G2_MULTIPLAYER_SYNC);
+            gfx_draw_sprite(&dpi, ImageId(imgId), screenPos + ScreenCoordsXY{ 3, 11 });
+
+            // Draw number of players.
+            auto ft = Formatter();
+            ft.Add<int32_t>(network_get_num_visible_players());
+            DrawTextBasic(
+                &dpi, screenPos + ScreenCoordsXY{ 23, 1 }, STR_COMMA16, ft,
+                { COLOUR_WHITE | COLOUR_FLAG_OUTLINE, TextAlignment::RIGHT });
+        }
     }
-}
+};
 
 /**
- *
- *  rct2: 0x0066CD9C
+ * Creates the main game top toolbar window.
+ *  rct2: 0x0066B485 (part of 0x0066B3E8)
  */
-static void ToggleWaterWindow(rct_window* topToolbar, WidgetIndex widgetIndex)
+rct_window* WindowTopToolbarOpen()
 {
-    if ((input_test_flag(INPUT_FLAG_TOOL_ACTIVE)) && gCurrentToolWidget.window_classification == WindowClass::TopToolbar
-        && gCurrentToolWidget.widget_index == WIDX_WATER)
-    {
-        tool_cancel();
-    }
-    else
-    {
-        _landToolBlocked = false;
-        show_gridlines();
-        tool_set(*topToolbar, widgetIndex, Tool::WaterDown);
-        input_set_flag(INPUT_FLAG_6, true);
-        ContextOpenWindow(WindowClass::Water);
-    }
+    rct_window* window = WindowCreate<TopToolbar>(
+        WindowClass::TopToolbar, ScreenCoordsXY(0, 0), ContextGetWidth(), TOP_TOOLBAR_HEIGHT + 1,
+        WF_STICK_TO_FRONT | WF_TRANSPARENT | WF_NO_BACKGROUND);
+
+    window->widgets = window_top_toolbar_widgets;
+
+    WindowInitScrollWidgets(*window);
```

Comment:
Move this to the same constructor mentioned above, where it can become just `InitScrollWidgets()`